### ISSUE TITLE
Torrent data not deleting

### ISSUE THREAD

**Issue by mitle** [Novik/ruTorrent#2215]
Torrents are deleted from the list, but the data files remain on disk. Both automatic (ratio group rules) and manual (right click, delete with data) methods fail. No error message appears in the browser. Deleted torrents are not visible in history (However when deleting withoout removing files, when the torrent is deleted from the client, then it is producing an entry and is visible in history.)

The system is on an ubuntu server, both the user running rtorrent and the webserver (apache) has rights to write (delete) files.  

Tried enabling debug log in conf.php of erasedata plugin, but it generated no output at all.

I also tired adding the following to rtorrent.rc, this deletes the files, but the action is still not registered in the ruTorrent history. `method.set_key = event.download.erased,delete_erased,"execute=rm,-rf,--,$d.base_path="`
On Reddit it is mentioned that this can be dangerous, and so I do not want to use it. 

Can someone help me with this issue? Is this an ruTorrent issue at all or is it rtorrent misconfiguration?

Attached the config files:
[config.php.txt](https://github.com/Novik/ruTorrent/files/7441566/config.php.txt)
[rtorrent.rc.txt](https://github.com/Novik/ruTorrent/files/7441568/rtorrent.rc.txt)
[apache-rtorrent.conf.txt](https://github.com/Novik/ruTorrent/files/7441601/apache-rtorrent.conf.txt)

Sys  and version info:
Ubuntu 20.04.2 LTS
PHP 7.4.23 (cli) (built: Aug 26 2021 15:51:37) ( NTS )
Apache/2.4.41 (Ubuntu)
rtorrent 0.9.8/0.13.8.
ruTorrent 3.10

**Comment by antetna** [comment]
and how did you install ruTorrent -  where/how did you get  tarball/sources ?

**Comment by mitle** [comment]
downloaded the v3.10.zip from the releases from github (this repository), then extracted it to /var/www/ruTorrent and then changed ownership to www-data ( user running apche server).

**Comment by antetna** [comment]
Your .rtorrent.rc must be for  rtorrent \<= 0.9.6  and when used with 0.9.8 and ruTorrent 3.10 some things do not work. nothing wrong with ruTorrent 3.10 or rtorrent 0.9.8, you just need to fix your .rtorrent.rc

A)

try to comment out all lines starting with 

**execute.\***   
**method.\***   
 in .rtorrent.rc   
  
B)  check if this is right, "user" is name of the user that is running rtorrent ?  
schedule2 = init\_plugins, 10, 0, "execute2 = {sh,-c, /usr/bin/php /var/www/ruTorrent/php/initplugins.php user &}"

**Comment by mitle** [comment]
Thanks!! Now it is fixed. The username there must be the one that is used for basic auth, not the user running rtorrent.
Details below.

I tried fixing the B) alone first, since that was obviously wrong. I changed it to the user **running rtorrent**. Hovever that did not solve the problem. Then I commented out just this _schedule2 initplugins.php_ line and then the issue was resolved. I opened that file and started looking around, I searched for _argv_. The only use of it is setting the `$_SERVER['REMOTE_USER']`. That means the username I use for basic auth (I use basic authentication for ruTorrent) must be given there, not the user running rtorrent.
